### PR TITLE
feat: import service validate duplicates

### PR DESCRIPTION
**Pull request by kwasniew**
## About the changes

Validate if we have duplicate feature in import.

Also preparing for duplication removal in the enterprise service by extracting shared methods.

### CODE REVIEW

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 1]
```diff
@@ -163,23 +166,23 @@ export default class ExportImportService {
...
             ),
+            this.getDuplicateFeatures(dto),
         ]);
 
-        const errors = ImportValidationMessages.compileErrors(
-            dto.project,
-            unsupportedStrategies,
-            unsupportedContextFields || [],
-            [],
+        const errors = ImportValidationMessages.compileErrors({
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:174]: introduced object to make it more manageable

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 2]
```diff
@@ -192,24 +195,36 @@ export default class ExportImportService {
...
-            ),
+            this.verifyStrategies(dto),
+            this.verifyContextFields(dto),
+            this.importPermissionsService.verifyPermissions(dto, user, mode),
             this.verifyFeatures(dto),
         ]);
-        await this.createOrUpdateToggles(cleanedDto, user);
+    }
+
+    async importToggleLevelInfo(
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:211]: used by enterprise

File: src/lib/features/export-import-toggles/export-import-service.ts [hunk 3]
```diff
@@ -192,24 +195,36 @@ export default class ExportImportService {
         };
     }
 
-    async import(dto: ImportTogglesSchema, user: User): Promise<void> {
-        const cleanedDto = await this.cleanData(dto);
-
+    async importVerify(
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/export-import-service.ts:198]: used by enterprise

File: src/lib/features/export-import-toggles/import-validation-messages.ts [hunk 4]
```diff
@@ -46,35 +59,28 @@ export class ImportValidationMessages {
                 ),
             });
         }
-        if (segments.length > 0) {
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/import-validation-messages.ts:49]: moved to enterprise

File: src/lib/features/export-import-toggles/import-validation-messages.ts [hunk 5]
```diff
@@ -46,35 +59,28 @@ export class ImportValidationMessages {
...
+        if (otherProjectFeatures.length > 0) {
             errors.push({
-                message:
-                    'We detected the following segments in the import file that need to be created first:',
-                affectedItems: segments,
+                message: `You cannot import a features that already exist in other projects. You already have the following features defined outside of project ${projectName}:`,
+                affectedItems: otherProjectFeatures,
             });
         }
-        if (changeRequestExists) {
```
Comment (kwasniew) [inline src/lib/features/export-import-toggles/import-validation-messages.ts:56]: moved to enterprise

**Review by sjaanus — APPROVED** [review]

**Pull request merged**